plugins/inputs/couchbase/couchbase.go

```diff
@@ -70,34 +70,34 @@ func (cb *Couchbase) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
-func (cb *Couchbase) gatherServer(acc telegraf.Accumulator, addr string, pool *couchbaseClient.Pool) error {
-	if pool == nil {
-		client, err := couchbaseClient.Connect(addr)
-		if err != nil {
-			return err
-		}
+func (cb *Couchbase) gatherServer(acc telegraf.Accumulator, addr string) error {
+	escapedAddr := regexpURI.ReplaceAllString(addr, "${1}")
 
-		// `default` is the only possible pool name. It's a
-		// placeholder for a possible future Couchbase feature. See
-		// http://stackoverflow.com/a/16990911/17498.
-		p, err := client.GetPool("default")
-		if err != nil {
-			return err
-		}
-		pool = &p
+	client, err := couchbaseClient.Connect(addr)
+	if err != nil {
+		return err
+	}
+
+	// `default` is the only possible pool name. It's a
+	// placeholder for a possible future Couchbase feature. See
+	// http://stackoverflow.com/a/16990911/17498.
+	pool, err := client.GetPool("default")
+	if err != nil {
+		return err
 	}
+	defer pool.Close()
```

Comment:
why do we close the pool and not the connection? Also shouldn't we be reusing both?

Reply:
Ideally, we would be reusing both. However, if we reuse the pool, there is no way to refresh the bucket map, so we wouldn't be able to get metrics for new buckets, and we would start erroring if a bucket gets deleted. 

We don't close the connection because there isn't a method to do that.